wethegit/masher
# feature: v1

# Goal

Masher V1!
This initial work is mostly conversion and refactoring. A lot of the logic was kept intact with some improvements here and there.

## Notes

### Timing
I am not able to spend much more time refactoring this as we need V1 ready to continue work on Corgi and Component Library, so please focus your test on potential bugs and incompatibilities.

With the hardest part in place we can keep track of changes and features in Issues from here on out.

### Picture component

I added the `Picture` component to the repo with the intention of releasing it soon. I have a ticket create for it so we can keep track of that feature #2 but please ignore it for now.

## Testing

You don't need a Corgi project to test this.

1. Clone this repo and checkout `feature/v1`
2. Make sure you are running the correct node version and `npm install`
3. `npm run build`
4. `npm link`

You are now ready to test
1. `mkdir my-test`
5. `cd my-test`
6. `npm init -y`
7. `npm link @wethegit/masher`
8. `mkdir -p src/images/defined/`
9. Add a few images to the `defined` directory created above
10. Run the masher `npx @wethegit/masher`
11. For watching for changes `npx @wethegit/masher --watch`

After running the masher you should see the compressed images in a `public/_images/` directory. You can change the input and output directories inside the `masher.config.json` file gets created after you ran it at least once.
